pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -167,6 +167,10 @@ func (mixerplugin) OnInboundListener(in *plugin.InputParams, mutable *plugin.Mut
 		attrs["context.proxy_version"] = attrStringValue(vs)
 	}
 
+	if meshID, found := in.Node.Metadata[model.NodeMetadataMeshID]; found {
+		attrs["destination.mesh.id"] = attrStringValue(meshID)
```

Comment:
curious: why `destination.mesh.id` and not `context.mesh.id` or similar ?  is there a difference between `source` and `destination` meshes?

Reply:
That was the outcome of the discussion between @djazayeri and @costinm . Note that we also only report inbound telemetry for destination.

Reply:
Is there a reason to only report inbound?

Should we be setting source.mesh.id ?

Reply:
The reason for source/destination instead of context is to design for the day when mesh federation is implemented (and source/destination might differ).
We decided to defer having source.mesh.id until federation happens +/- we can strongly state that another mesh id besides this mesh is correct.

Reply:
I guess that makes some sense, but I'm worried about mid-term confusion (and the upgrade experience). We don't really have a timeline for mesh federation, as I understand it. And having the unbalance between `destination.mesh.id` existing and no `source.mesh.id` seems like it could trip people up.

Introducing `context.mesh.id` (and then deprecating when mesh federation support was added) would have been my preferred approach, as I feel like there is going to be a lot of other things that we have to figure out and shuffle around for federation scenarios -- and in the meantime it leads to more questions.

But it sounds like this decision has been signed off on and agreed to by multiple parties, so I won't stop progress. The future role of attributes is still largely unresolved as is.